packages/client-api/src/providers/data/create-data-provider.ts

```diff
@@ -0,0 +1,65 @@
+import { type Owner } from 'solid-js';
+import { createStore } from 'solid-js/store';
+
+import type { DataProviderConfig } from '~/user-config';
+
+export interface DataVariables { }
+
+export async function createDataProvider(
+  config: DataProviderConfig,
+  owner: Owner,
+) {
+  const [dataVariables, setDataVariables] =
+    createStore<DataVariables>(getDataVariables());
+
+  function getDataVariables() {
+    return {};
+  }
+
+  const bitUnits = ['b', 'Kb', 'Mb', 'Gb', 'Tb', 'Pb', 'Eb', 'Zb', 'Yb'];
+  const byteCommonUnits = ['B', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB'];
+  const byteIECUnits = [' B', 'KiB', 'MiB', 'GiB', 'TiB', 'PiB', 'EiB', 'ZiB', 'YiB'];
+
+  function convertBytes(bytes: number, decimals: number, minUnit: string) {
```

Comment:
Could we remove the `minUnit` arg and use the smallest unit that makes sense (e.g. `8000` -> `8Kb`, `8000000` -> `8Mb`)?
```ts
// would these names make sense? 
export enum DataUnit {
  BITS = 'bits',
  SI_BYTES = 'si_bytes',
  IEC_BYTES = 'iec_bytes',
}

function convertBytes(bytes: number, decimals: number = 0, unitType: DataUnit = DataUnit.BITS) {
  ...
}
```

Reply:
@lars-berger I think that defaulting to `KB/KiB` makes sense, I would never care to see bits/bytes

I changed to minUnit to unitType, but I am not sure how to expose the DataUnit enum. I am not used to typescript? I know I am missing something.

```javascript
// obviously does not work since the template does not know this enum type
{{util.convertBytes(1024, 2, DataUnit.SI_BYTES)}}
```

Reply:
We do the same for the weather provider (ref the `WeatherStatus` enum and its usage in the sample config)

Reply:
@lars-berger I was not aware. It makes sense.